### PR TITLE
Issue/sling 8038

### PR DESCRIPTION
**Pull request by DominikSuess**

### CODE REVIEW

File: src/main/java/org/apache/sling/feature/maven/mojos/ArtifactsMojo.java [hunk 1]
```diff
@@ -38,8 +38,7 @@
 
 @Mojo(
     name = "collect-artifacts",
-    defaultPhase = LifecyclePhase.PACKAGE,
-    requiresDependencyResolution = ResolutionScope.TEST,
+    requiresProject = false,
```
Comment (simonetripodi) [inline src/main/java/org/apache/sling/feature/maven/mojos/ArtifactsMojo.java]: good catch, thanks! 👍

File: src/main/java/org/apache/sling/feature/maven/mojos/AbstractRepositoryMojo.java [hunk 2]
```diff
@@ -47,7 +47,7 @@
     /**
      * The directory to store the artifacts into.
      */
-    @Parameter(defaultValue = "artifacts")
+    @Parameter(defaultValue = "artifacts", property = "repositoryDir")
```
[the remark this answers is not in the document]
Comment (DominikSuess) [reply src/main/java/org/apache/sling/feature/maven/mojos/AbstractRepositoryMojo.java]: @simonetripodi  this is intentional as I want to be able to fully control the target directory of the download target when not executing within a maven project context.

File: src/main/java/org/apache/sling/feature/maven/mojos/AbstractRepositoryMojo.java [hunk 3]
```diff
@@ -65,7 +65,12 @@
 
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
-        final File artifactDir = new File(this.project.getBuild().getDirectory(), repositoryDir);
+        final File artifactDir ;
```
[the remark this answers is not in the document]
Comment (DominikSuess) [reply src/main/java/org/apache/sling/feature/maven/mojos/AbstractRepositoryMojo.java]: @simonetripodi  the intention was to get rid of a project dir and make this independent of a project - following patterns of other maven plugins I detect the unreplaced project build dir to handle this case. If triggered in a non project folder the target folder can be defined relative to execution dir or absolute.

**Comment by DominikSuess** [comment]
@cziegeler  - replacing with PR-15 with updated commit sequence (moving the abstraction into own subsequent commit for ease of reuse.

**Pull request closed without merging**